File: recipe-service/src/main/java/com/demeter/recipeservice/dto/RecipeRequest.java
package com.demeter.recipeservice.dto;

import com.demeter.recipeservice.model.Photo;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecipeRequest {
    private String name;
    private Integer servingSize;
    private String author;
    private BigDecimal price;
    private List<IngredientRequest> ingredientList;
    private List<StepRequest> method;
    private Integer preparationTime;
    private Integer cookingTime;
    private Integer restingTime;
    private Photo photo;
}

File: demeter-ui/src/app/addrecipe/recipeRequest.ts
import {Ingredient} from "./ingredient";
import {Step} from "./step";
import {UploadPhotoResponse} from "./uploadPhotoResponse";

export class RecipeRequest {

  constructor(
    public name: string,
    public author: string,
    public servingSize: number,
    public ingredientList: Ingredient[],
    public method: Step[],
    public price: number,
    public preparationTime: number,
    public cookingTime: number,
    public restingTime: number,
    public photo: UploadPhotoResponse
  ){

  }
}
